library/objective-c/EnvoyEngineImpl.m

```diff
@@ -66,40 +62,51 @@ static envoy_headers toNativeHeaders(EnvoyHeaders *headers) {
   return (envoy_headers){length, header_array};
 }
 
+static const void* ios_http_filter_init(const void *context) {
+  EnvoyHTTPFilterFactory *filterFactory = (__bridge EnvoyHTTPFilterFactory *)context;
+  EnvoyHTTPFilter *filter = filterFactory.create();
+  return CFBridgingRetain(filter);
+}
+
 static envoy_filter_headers_status
-ios_http_filter_on_request_headers(envoy_headers headers, bool end_stream, void *context) {
+ios_http_filter_on_request_headers(envoy_headers headers, bool end_stream, const void *context) {
   // TODO(goaway): optimize unmodified case
-  ios_http_filter_context *c = (ios_http_filter_context *)context;
-  if (c->filter.onRequestHeaders == nil) {
+  EnvoyHTTPFilter *filter = (__bridge EnvoyHTTPFilter *)context;
+  if (filter.onRequestHeaders == nil) {
     return (envoy_filter_headers_status){/*status*/ kEnvoyFilterHeadersStatusContinue,
                                          /*headers*/ headers};
   }
 
   EnvoyHeaders *platformHeaders = to_ios_headers(headers);
   release_envoy_headers(headers);
   // TODO(goaway): consider better solution for compound return
-  NSArray *result = c->filter.onRequestHeaders(platformHeaders, end_stream);
+  NSArray *result = filter.onRequestHeaders(platformHeaders, end_stream);
   return (envoy_filter_headers_status){/*status*/ [result[0] intValue],
                                        /*headers*/ toNativeHeaders(result[1])};
 }
 
 static envoy_filter_headers_status
-ios_http_filter_on_response_headers(envoy_headers headers, bool end_stream, void *context) {
+ios_http_filter_on_response_headers(envoy_headers headers, bool end_stream, const void *context) {
   // TODO(goaway): optimize unmodified case
-  ios_http_filter_context *c = (ios_http_filter_context *)context;
-  if (c->filter.onResponseHeaders == nil) {
+  EnvoyHTTPFilter *filter = (__bridge EnvoyHTTPFilter *)context;
+  if (filter.onResponseHeaders == nil) {
     return (envoy_filter_headers_status){/*status*/ kEnvoyFilterHeadersStatusContinue,
                                          /*headers*/ headers};
   }
 
   EnvoyHeaders *platformHeaders = to_ios_headers(headers);
   release_envoy_headers(headers);
   // TODO(goaway): consider better solution for compound return
-  NSArray *result = c->filter.onResponseHeaders(platformHeaders, end_stream);
+  NSArray *result = filter.onResponseHeaders(platformHeaders, end_stream);
   return (envoy_filter_headers_status){/*status*/ [result[0] intValue],
                                        /*headers*/ toNativeHeaders(result[1])};
 }
 
+static void ios_http_filter_release(const void *context) {
+  CFRelease(context);
+  return;
```

Comment:
Is this necessary?

Reply:
The `return` statement? Technically no, but I guess I tend to err on the side of being explicit.